### PR TITLE
net: add uv_ip_name to get name from sockaddr

### PR DESCRIPTION
**Pull request by duskmoon314**
discussion #3365 

# feature

uv_ip_name is a kind of wrapper of uv_ip4_name and uv_ip6_name which can be used after getaddrinfo to get the IP name directly from addrinfo.ai_addr.

# notice

I haven't found a test just for `uv_ip*_name`, so I created a test called `ip_name` and set up a very simple case. Maybe a better way is to generate all the legal and illegal strings and feed them into these functions.

### CODE REVIEW

File: test/test-ip-name.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include "uv.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <string.h>
+
+
+TEST_IMPL(ip_name) {
+    char dst[INET6_ADDRSTRLEN];
+    struct sockaddr *addr = (struct sockaddr*)malloc(sizeof(struct sockaddr_in6));
```
Comment (bnoordhuis) [inline test/test-ip-name.c:31]: Not necessary to heap-allocate here, just stack-allocate it, maybe through a union.

Note that this currently leaks the memory.

File: test/test-ip-name.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    struct sockaddr_in *addr4 = (struct sockaddr_in*)addr;
+    struct sockaddr_in6 *addr6 = (struct sockaddr_in6*)addr;
+
+    /* test ip4_name */
+    ASSERT(0 == uv_ip4_addr("192.168.0.1", TEST_PORT, addr4));
```
Comment (bnoordhuis) [inline test/test-ip-name.c:36]: Can you use the new-style `ASSERT_EQ(0, ...)` macro? Thanks.

**Comment by duskmoon314** [comment]
@bnoordhuis I use a union now, but I'm not sure am I used it correctly. Please check it.

File: test/test-ip-name.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    ASSERT_EQ(0, uv_ip6_name(addr6, dst, INET6_ADDRSTRLEN));
+    ASSERT_EQ(0, strcmp("fe80::2acf:daff:fedd:342a", dst));
+    
+    ASSERT_EQ(0, uv_ip_name(addr, dst, INET6_ADDRSTRLEN));
+    ASSERT_EQ(0, strcmp("fe80::2acf:daff:fedd:342a", dst));
+
+    /* test other sa_family */
+    addr->sa_family = AF_UNIX;
+    // size is not concerned
+    ASSERT(UV_EAFNOSUPPORT == uv_ip_name(addr, dst, INET6_ADDRSTRLEN));
```
Comment (bnoordhuis) [inline test/test-ip-name.c:61]:
```suggestion
    /* size is not a concern here */
    ASSERT_EQ(UV_EAFNOSUPPORT, uv_ip_name(addr, dst, INET6_ADDRSTRLEN));
```

**Review by bnoordhuis — APPROVED** [review]
LGTM, thanks!

**Pull request merged**